src/formulas/compiler.ts

```diff
@@ -316,43 +318,53 @@ function formulaArguments(tokens: Token[]) {
  * Check if arguments are supplied in the correct quantities
  */
 function assertEnoughArgs(ast: ASTFuncall) {
-  const nbrArg = ast.args.length;
+  const nbrArgSupplied = ast.args.length;
   const functionName = ast.value.toUpperCase();
   const functionDefinition = functions[functionName];
+  const { nbrArgRepeating, minArgRequired } = functionDefinition;
 
-  if (nbrArg < functionDefinition.minArgRequired) {
+  if (nbrArgSupplied < minArgRequired) {
     throw new BadExpressionError(
       _t(
-        "Invalid number of arguments for the %s function. Expected %s minimum, but got %s instead.",
-        functionName,
-        functionDefinition.minArgRequired.toString(),
-        nbrArg.toString()
+        "Invalid number of arguments for the %(functionName)s function. Expected %(minArgRequired)s minimum, but got %(nbrArgSupplied)s instead.",
+        {
+          functionName: functionName,
+          minArgRequired: minArgRequired.toString(),
+          nbrArgSupplied: nbrArgSupplied.toString(),
+        }
       )
     );
   }
 
-  if (nbrArg > functionDefinition.maxArgPossible) {
+  if (nbrArgSupplied > functionDefinition.maxArgPossible) {
     throw new BadExpressionError(
       _t(
-        "Invalid number of arguments for the %s function. Expected %s maximum, but got %s instead.",
-        functionName,
-        functionDefinition.maxArgPossible.toString(),
-        nbrArg.toString()
+        "Invalid number of arguments for the %(functionName)s function. Expected %(maxArgPossible)s maximum, but got %(nbrArgSupplied)s instead.",
+        {
+          functionName: functionName,
+          maxArgPossible: functionDefinition.maxArgPossible.toString(),
+          nbrArgSupplied: nbrArgSupplied.toString(),
+        }
       )
     );
   }
 
-  const repeatableArgs = functionDefinition.nbrArgRepeating;
-  if (repeatableArgs > 1) {
-    const unrepeatableArgs = functionDefinition.args.length - repeatableArgs;
-    const repeatingArgs = nbrArg - unrepeatableArgs;
-    if (repeatingArgs % repeatableArgs !== 0) {
+  if (nbrArgRepeating > 1) {
+    const nbrValueRepeating =
+      nbrArgRepeating * Math.floor((nbrArgSupplied - minArgRequired) / nbrArgRepeating);
+    const nbrValueRemaining =
+      nbrArgSupplied - minArgRequired - nbrValueRepeating - functionDefinition.nbrArgOptional;
+
+    if (nbrValueRemaining > 0) {
       throw new BadExpressionError(
         _t(
-          "Invalid number of arguments for the %s function. Expected all arguments after position %s to be supplied by groups of %s arguments",
-          functionName,
-          unrepeatableArgs.toString(),
-          repeatableArgs.toString()
+          "Invalid number of arguments for the %(functionName)s function. Repeatable arguments are expected to be supplied by groups of %(nbrArgRepeating)s argument(s) with maximum %(nbrArgOptional)s optional argument(s), but got %(nbrValueRemaining)s argument(s) too many.",
```

Comment:
I think it would still be a valid sentence if we just remove the `argument(s)` ? I would be better if we didn't have this (s). It's not a particularly good UX to see "0 argument(s)"